browser/browser_context_keyed_service_factories.cc

```diff
@@ -111,10 +106,6 @@ void EnsureBrowserContextKeyedServiceFactoriesBuilt() {
   PermissionLifetimeManagerFactory::GetInstance();
 
   skus::SkusServiceFactory::GetInstance();
-
-#if BUILDFLAG(ENABLE_PLAYLIST)
-  playlist::PlaylistServiceFactory::GetInstance();
```

Comment:
We don't need to instantiate playlist service factory in advance. Let's defer.

Reply:
Nice catch! this was the culprit! 

BTW with all due respect, I'd like to talk about a thread.

> no factories are registered after Ensure*KeyedServiceFactoriesBuilt() is called [1]

> All factories must have been created at least once before the context is created in order to work correctly  [2]

[1] https://bugs.chromium.org/p/chromium/issues/detail?id=1150733#c7
[2] https://source.chromium.org/chromium/chromium/src/+/main:components/keyed_service/core/keyed_service_base_factory.h;l=26?q=components%2Fkeyed_service%2Fcore%2Fkeyed_service_base_factory.h&ss=chromium


If I understand this thread correctly, All factories should be created before this `Ensure...` method, i.e. we shouldn't defer factory creation later than this function. Especially, there's no other factories that depend on this factory, we should manually instantiate it at least once.

Also, I think if we don't call this here, `RegisterProfilePrefs()` may not be called by DependencyManager.